paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/PickFilesForCloneBuilder.java

```diff
@@ -0,0 +1,245 @@
+                        .map(
+                                entry ->
+                                        new CloneFileInfo(
+                                                entry.getValue().getLeft(),
+                                                entry.getValue().getRight(),
+                                                entry.getKey().startsWith(SNAPSHOT_PREFIX)
+                                                        || entry.getKey().startsWith(TAG_PREFIX)))
+                        .collect(Collectors.toList());
+            default:
+                throw new UnsupportedOperationException("Unknown cloneType : " + cloneType);
+        }
+    }
+
+    /**
+     * @param fileNames all fileNames of a snapshot or tag
+     * @param tableAllFiles all files of a table. path is the relative path
+     * @param snapshotOrTagFile filter by snapshot or tag file
+     * @return
+     */
+    private List<CloneFileInfo> getFilePathsForSnapshotOrTag(
+            List<String> fileNames,
+            Map<String, Pair<Path, Long>> tableAllFiles,
+            Predicate<String> snapshotOrTagFile) {
+        List<CloneFileInfo> cloneFileInfos = new ArrayList<>();
+
+        for (String fileName : fileNames) {
+            Pair<Path, Long> filePathAndSize = tableAllFiles.get(fileName);
+            checkNotNull(
+                    filePathAndSize,
+                    "Error this is a bug, please report. fileName is "
+                            + fileName
+                            + ", tableAllFiles is : "
+                            + tableAllFiles);
+            cloneFileInfos.add(
+                    new CloneFileInfo(
+                            filePathAndSize.getKey(),
+                            filePathAndSize.getValue(),
+                            snapshotOrTagFile.test(fileName)));
+        }
+        return cloneFileInfos;
+    }
+
+    /**
+     * Get all files of a table.
+     *
+     * @return fileName -> (relative path, file size)
+     */
+    private Map<String, Pair<Path, Long>> getTableAllFilesExcludeTableRoot() {
+        Map<String, Pair<Path, Long>> tableAllFiles =
+                PickFilesUtil.getTableAllFiles(sourceTableRoot, partitionNum, fileIO);
+
+        return tableAllFiles.entrySet().stream()
+                .collect(
+                        Collectors.toMap(
+                                Map.Entry::getKey,
+                                e ->
+                                        Pair.of(
+                                                getPathExcludeTableRoot(e.getValue().getKey()),
+                                                e.getValue().getRight())));
+    }
+
+    /**
+     * cut table root path of absolutePath.
+     *
+     * @param absolutePath
+     * @return the relative path
+     */
+    private Path getPathExcludeTableRoot(Path absolutePath) {
+        String fileAbsolutePath = absolutePath.toUri().getPath();
```

Comment:
```suggestion
        String fileAbsolutePath = absolutePath.toUri().toString();
```

`getPath` does not include scheme, authority and such.

Reply:
done